1175/en.md

```diff
@@ -0,0 +1,138 @@
+# LOJ 1175 - Jane and the Frost Giants
+
+## Summary
+
+**Input:**
+
+A 2D grid representing the maze, where each square is either empty, blocked by an obstacle, or contains fire lit by the *Frost Giants*, except one cell which is the starting location of *Jane* in the maze. Both *Jane* and the fire take one minute to move to an adjacent cell (horizontally or vertically).
+
+**Output:**
+
+A boolean value indicating whether *Jane* can escape the maze before the fire reaches her. If *Jane* can escape, the minimum number of minutes it takes for her to escape from the maze.
+
+**Note:** *Jane* can escape from any square that borders the edge of the maze.
+
+## Prerequisite
+**BFS** (Bredth First Search): https://cp-algorithms.com/graph/breadth-first-search.html
+
+## Solution
+If we forget about the movable fire cells that grows every minute, then the problem reduces to calculating how many minutes it takes to reach a cell without going through an obstacle, which can be done easily with BFS alone.
+
+ However, taking the fire cells into account, we can think of them as movable osbtacles that expand to the adjacent cells every minute causing *Jane* not to move through them. Before Jane moves to any of her adjacent cells, the fire cells must move to their adjacent cells, because Jane can only move through that cell if the fire hasn't lit up in that very moment. 
+
+## Complexity
+- Time Complexity: O(T * $R*C$).
+- Memory Complexity: O($R * C$).
+
+## Code
+
+### C++
+
+```cpp
+#include <bits/stdc++.h>
+
+using namespace std;
+
+typedef pair <int, int> pii;
+
+
+const int INF = numeric_limits <int>:: max(); // Unreachable
+
+int r, c;
+vector <string> grid;
+
+// An Efficient (and quite common) Way to Navigate Grid Problems: https://codeforces.com/blog/entry/78827
+
+const int dr[] = {-1, 0, 1, 0};
+const int dc[] = {0, 1, 0, -1};
+
+inline bool valid(int x, int y) {
+    return x >= 0 && x < r && y >= 0 && y < c && grid[x][y] != '#'; // 0-based index grid
+}
+
+struct Cell {
+    bool flag; // Indicating whther this cell is on fire or not
+    int x, y; // Row and column respectively 
+
+    Cell() {} // Default constrcutor
+    Cell(bool flag, int x, int y) : flag(flag), x(x), y(y) {}
+};
+
+int main() {
+    
+    // For fast I/O
+    ios_base::sync_with_stdio(false);
+    cin.tie(nullptr);
+
+    int t;
+    cin >> t;
+
+    for(int ts = 1; ts <= t; ++ts) {
+        cin >> r >> c;
+
+        grid.resize(r);
+        
+        pii source;
+        queue <Cell> Q;
+        vector <vector <int>> dist(r, vector <int> (c, INF)); // Minimum minutes needed to move to cell[i][j] from starting cell
```

Comment:
Let's add comments on separate lines, not at right. I saw a few other occurrences, let's please change those as well.

```suggestion
        // Minimum minutes needed to move to cell[i][j] from starting cell
        vector <vector <int>> dist(r, vector <int> (c, INF));
```